Assets/LeapMotionModules/UI/Interaction/InteractionManager.cs

```diff
@@ -0,0 +1,293 @@
+    /// one Unity unit to correspond to one meter.
+    /// </summary>
+    public float SimulationScale { get { return _providerScale; } }
+
+    private InteractionHand[] _interactionHands = new InteractionHand[2];
+    private HashSet<InteractionBehaviourBase> _interactionBehaviours = new HashSet<InteractionBehaviourBase>();
+    private Dictionary<Rigidbody, InteractionBehaviourBase> _rigidbodyRegistry = new Dictionary<Rigidbody, InteractionBehaviourBase>();
+    public Dictionary<Rigidbody, InteractionBehaviourBase> RigidbodyRegistry { get { return _rigidbodyRegistry; } }
+
+    /// <summary> Stores data for implementing Soft Contact for InteractionHands. </summary>
+    [NonSerialized]
+    public List<PhysicsUtility.SoftContact> _softContacts = new List<PhysicsUtility.SoftContact>(80);
+    /// <summary> Stores data for implementing Soft Contact for InteractionHands. </summary>
+    [NonSerialized]
+    public Dictionary<Rigidbody, PhysicsUtility.Velocities> _softContactOriginalVelocities = new Dictionary<Rigidbody, PhysicsUtility.Velocities>(5);
+
+    void OnValidate() {
+      contactOrGraspingEnabled = enableContact || enableGrasping;
+
+      if (!Application.isPlaying && _autoGenerateLayers) {
+        AutoGenerateLayers();
+      }
+    }
+
+    /// <summary> Often, only one InteractionManager is necessary per Unity scene.
+    /// This property will contain that InteractionManager as soon as its Awake()
+    /// method is called. Using more than one InteractionManager is valid, but be
+    /// sure to assign the InteractionBehaviour's desired manager appropriately.
+    /// </summary>
+    /// 
+    /// <remarks> By default, this static property contains the first InteractionManager
+    /// that has had its Awake() method called in the current scene. If an
+    /// InteractionBehaviourBase does not have a non-null interactionManager by the
+    /// time it has Start() called, it will default to using the InteractionManager
+    /// referenced here.
+    /// 
+    /// If you have multiple InteractionManagers in your scene, you should be sure to
+    /// assign InteractionBehaviours' managers appropriately. If you instantiate an
+    /// InteractionBehaviour at runtime, you should assign its InteractionManager
+    /// right after you instantiate it. </remarks>
+    public static InteractionManager singleton { get; set; }
+
+    void Awake() {
+      if (InteractionManager.singleton == null) singleton = this;
+
+      Provider = Hands.Provider;
+
+      _interactionHands[0] = new InteractionHand(this, () => { return Hands.FixedLeft; }, Chirality.Left, WorldHoverActivationRadius, WorldTouchActivationRadius);
+      _interactionHands[1] = new InteractionHand(this, () => { return Hands.FixedRight; }, Chirality.Right, WorldHoverActivationRadius, WorldTouchActivationRadius);
+
+      if (_autoGenerateLayers) {
+        AutoGenerateLayers();
+        AutoSetupCollisionLayers();
+      }
+    }
+
+    void Start() {
```

Comment:
Eventually going to need correct enable/disable handling for when a manager is enabled/disabled at runtime.  This will include things like letting all currently grasped objects know they have been released.